File: smash-ranks-android/app/src/main/java/com/garpr/android/models/PlayerMatchesBundle.java
package com.garpr.android.models;

import android.os.Parcel;
import android.os.Parcelable;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

public class PlayerMatchesBundle implements Parcelable {

    @SerializedName("full_player")
    private FullPlayer mFullPlayer;

    @Nullable
    @SerializedName("matches_bundle")
    private MatchesBundle mMatchesBundle;


    @NonNull
    public FullPlayer getFullPlayer() {
        if (mFullPlayer == null) {
            throw new RuntimeException("mFullPlayer is null");
        }

        return mFullPlayer;
    }

    @Nullable
    public MatchesBundle getMatchesBundle() {
        return mMatchesBundle;
    }

    public boolean hasMatchesBundle() {
        return mMatchesBundle != null && mMatchesBundle.hasMatches();
    }

    public void setFullPlayer(@NonNull final FullPlayer fullPlayer) {
        mFullPlayer = fullPlayer;
    }

    public void setMatchesBundle(@Nullable final MatchesBundle matchesBundle) {
        mMatchesBundle = matchesBundle;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(final Parcel dest, final int flags) {
        dest.writeParcelable(mFullPlayer, flags);
        dest.writeParcelable(mMatchesBundle, flags);
    }

    public static final Creator<PlayerMatchesBundle> CREATOR = new Creator<PlayerMatchesBundle>() {
        @Override
        public PlayerMatchesBundle createFromParcel(final Parcel source) {
            final PlayerMatchesBundle pmb = new PlayerMatchesBundle();
            pmb.mFullPlayer = source.readParcelable(FullPlayer.class.getClassLoader());
            pmb.mMatchesBundle = source.readParcelable(MatchesBundle.class.getClassLoader());
            return pmb;
        }

        @Override
        public PlayerMatchesBundle[] newArray(final int size) {
            return new PlayerMatchesBundle[size];
        }
    };

}

File: smash-ranks-android/app/src/main/java/com/garpr/android/models/PlayerMatchesBundle.kt
package com.garpr.android.models

import android.os.Parcel
import android.os.Parcelable
import com.garpr.android.extensions.createParcel
import com.garpr.android.extensions.requireParcelable

import com.google.gson.annotations.SerializedName

data class PlayerMatchesBundle(
        @SerializedName("full_player") val fullPlayer: FullPlayer,
        @SerializedName("matches_bundle") val matchesBundle: MatchesBundle? = null
) : Parcelable {

    companion object {
        @JvmField
        val CREATOR = createParcel {
            PlayerMatchesBundle(
                it.requireParcelable(FullPlayer::class.java.classLoader),
                it.readParcelable(MatchesBundle::class.java.classLoader)
            )
        }
    }

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeParcelable(fullPlayer, flags)
        dest.writeParcelable(matchesBundle, flags)
    }

}
